examples/plot_text.py

```diff
@@ -0,0 +1,165 @@
+"""DiCoDiLe on text images
+==============================
+
+This example illlustrates pattern recovery on a noisy text image using
+DiCoDiLe algorithm.
+
+"""  # noqa
+import matplotlib.pyplot as plt
+import numpy as np
+
+from dicodile import dicodile
+from dicodile.data.images import fetch_pami
+from dicodile.update_d.update_d import tukey_window
+from dicodile.utils import check_random_state
+from dicodile.utils.csc import reconstruct
+from dicodile.utils.dictionary import init_dictionary, prox_d
+from dicodile.utils.viz import display_dictionaries
+
+
+###############################################################################
+# We will first load PAMI image generated from a text of **5000**
+# characters drawn uniformly from the **4** letters **P** **A** **M**
+# **I** and 2 whitespaces and assign it to `X`.
+#
+# We will also load the images of the four characters used to generate
+# `X` and assign it to variable `D`.
+
+X_original, D = fetch_pami()
+
+
+###############################################################################
+# We will work on the copy `X` of the original image and we need to
+# reshape image data `X` and `D` to fit to expected signal shape of
+# `dicodile`:
+#
+# (n_channels, *sig_support)
+
+X = X_original.copy()
+
+X = X.reshape(1, *X.shape)
+D = D.reshape(4, 1, *D.shape[-2:])
+
+# pad `D`
+D = np.pad(D, [(0, 0), (0, 0), (4, 4), (4, 4)])
+
+
+###############################################################################
+# Let's display an extract of the original text image `X_original` and
+# all the images of characters from `D`.
+
+extract_x = X_original[190:490, 250:750]
+plt.axis('off')
+plt.imshow(extract_x, cmap='gray')
+
+display_dictionaries(D)
+
+###############################################################################
+# We add a Gaussian white noise with standard deviation σ std (X) and σ = 3
+# to `X`.
+
+std = 3
+rng = check_random_state(None)
```

Comment:
```suggestion
rng = np.random.default_rng(None)
```

Reply:
Following this change, I changed 
`X += std * X.std() * rng.randn(*X.shape)` and `noise = noise_level_ * rng.randn(*D_init.shape)` as
`X += std * X.std() * rng.random(X.shape)` and `noise = noise_level_ * rng.random(D_init.shape)` respectively.

This resulted in a noisy initial image and randomly generated dictionary from it as:

![noisy](https://user-images.githubusercontent.com/2099645/110761626-40bc2a00-8250-11eb-9fb5-520e8565ed64.png)

Then the computation lasted 145 min and the result is as follows:
![dict](https://user-images.githubusercontent.com/2099645/110761918-8e389700-8250-11eb-8c5a-edbf2fb45991.png)


![result](https://user-images.githubusercontent.com/2099645/110761932-9264b480-8250-11eb-928b-c52b7d7d4c3c.png)

[Here](https://numpy.org/doc/stable/reference/random/generated/numpy.random.RandomState.randn.html#numpy.random.RandomState.randn) I see that `RandomState.rand` creates samples  from the “standard normal” distribution. So I tried with:

`X += std * X.std() * rng.standard_normal(X.shape)` and `noise = noise_level_ * rng.standard_normal(D_init.shape)` and this results in:

![last](https://user-images.githubusercontent.com/2099645/110762414-0ef79300-8251-11eb-81e0-f9415d57e37d.png)

I wanted to verify with you that this is the way to go.

May be we can update random generation using new Generator instead of  [RandomState](https://numpy.org/doc/stable/reference/random/index.html).